Gruntfile.js

```diff
@@ -73,16 +80,270 @@ module.exports = function (grunt) {
+                versionSpecs[index] = 0;
+            }
+            var newVersionNumber = versionSpecs.join(".").toString(),
+                newVersion = newVersionNumber;
+            if (isDev) {
+                newVersion = newVersion + "-dev";
+            }
+
+            // Update the JSON files with the latest version.
+            var jsonFileData = grunt.file.readJSON(jsonFile);
+            jsonFileData.version = newVersion;
+            grunt.file.write(jsonFile, JSON.stringify(jsonFileData));
+        });
+
+        // Prettify the concatenated JSON files.
+        grunt.task.run("jsonprettify");
+        done();
+    });
+
+    /**
+     * TODO: Create a grunt plugin replacing the current "github-release" task.
+     * (might be of interest to the community)
+     */
+    grunt.registerTask("githubRelease", "Task for automated GitHub releases.", function () {
+        // Reference to the function triggered when the asynchronous task is completed.
+        var done = this.async();
+
+        // Verify that the authentication token has been supplied correctly.
+        if (!process.env.GITHUB_ACCESS_TOKEN) {
+            grunt.log.error("GitHub access token cannot be found.");
+            return;
         }
+        grunt.log.subhead("Connecting...");
+
+        var apiUrl = "https://api.github.com",
+            uploadUrl = "https://uploads.github.com";
+
+        var taskOptions = this.options(),
+            repo = taskOptions.repo,
+            responseStatus = null;
+
+        var releaseUrl = apiUrl + "/repos/" + repo + "/releases";
+        fetch(releaseUrl, { headers: { "Authorization": "Token " + process.env.GITHUB_ACCESS_TOKEN } })
+            .then(function (response) {
+                responseStatus = response.status;
+
+                // Verify that the repository is accessible.
+                if (responseStatus !== 200) {
+                    if (responseStatus === 404) {
+                        throw new Error("Could not find the repository: https://github.com/" + taskOptions.owner + "/" + taskOptions.repo);
+                    }
+                    else {
+                        throw new Error("Could not connect to the repository.");
+                    }
+                }
+                grunt.log.ok("The connection has been established.");
+
+                // Create a release if the repository is accessible.
+                grunt.log.subhead("Creating new release \"" + taskOptions.tag_name + "\" on GitHub...");
+                createRelease(uploadReleaseAsset, taskOptions);
+            })
+            .catch(function (errorMessage) {    // eslint-disable-line
+                grunt.log.error(errorMessage);
+
+                // Indicate that the task has completed due to failure.
+                done();
+            });
+
+        // Creates a release in the repository.
+        var createRelease = function (callback, taskOptions) {   // eslint-disable-line
```

Comment:
Why did you have to disable ESLint here?

Reply:
If there is a good rationale, then at a minimum I would suggest disabling the specific rule with a comment nearby.

Reply:
Sorry, that was added by mistake. It can run absolutely fine without the rule. I'll remove it in the next commit.